sockeye/training.py

```diff
@@ -40,6 +41,16 @@
 logger = logging.getLogger(__name__)
 
 
+def global_norm(ndarrays):
+    # accumulate in a list, as asscalar is blocking and this way we can run the norm calculation in parallel.
+    norms = []
+    for arr in ndarrays:
+        if arr is None:
+            continue
+        norms.append(mx.nd.square(mx.nd.norm(arr)))
```

Comment:
would the parallelism be broken if a list comprehension would be used?

Reply:
I don't think so. I guess it makes sense to keep asscalar and the rest apart. I'll change it to a list comprehension.